obj.go

```diff
@@ -87,6 +87,26 @@ func (O *Object) SetAttribute(name string, data *Data) error {
 	return nil
 }
 
+// InitAttributes prepare all atributes for use the object as IN parameter
+func (O *Object) InitAttributes() error {
+	data := Data{}
+	for _, attr := range O.Attributes {
+		data.reset()
+		data.NativeTypeNum = attr.NativeTypeNum
+		data.ObjectType = attr.ObjectType
+		if attr.NativeTypeNum == C.DPI_NATIVE_TYPE_BYTES && attr.OracleTypeNum == C.DPI_ORACLE_TYPE_NUMBER {
+			var a [22]byte
+			C.dpiData_setBytes(data.dpiData, (*C.char)(unsafe.Pointer(&a[0])), 22)
```

Comment:
a is empty, so why are we setting that?
Ain't C.dpiData_getBytes is what you're looking for?

Reply:
Yes. C.dpiData_getBytes is the better function here